### ISSUE TITLE
Upgrade to using Python 3.10.4

### ISSUE THREAD

**Issue by lisagoodrich** [dtcenter/MET#2421]
## Describe the Task ##
Update any code necessary for upgrading to Python 3.10.4.  We are taking this action because WCOSS2 is only allowing Python 3.10.4 to be used for operational installations.  Therefore, we need to ensure that our next release uses Python 3.10.4.

**Please also update .readthedocs.yaml with 3.10.4**
Note that we decided NOT to upgrade to 3.10.4 since RTD supports up to 3.8.

### Time Estimate ###
*Estimate the amount of work required here.*
*Issues should represent approximately 1 to 3 days of work.*
**(Assignee please fill this in once you know.)**

### Sub-Issues ###
Consider breaking the task down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
Incorporate in a beta version as soon as possible to ensure there are no problems.

### Funding Source ###
Please ask Tara before starting on this work.

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [x] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issues
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

**Comment by JohnHalleyGotway** [comment]
I see that PR #2440 was merged and closed. However, I also see that `.readthedocs.yaml` still references 3.8:
```
python:
   version: 3.8
   install:
   - requirements: docs/requirements.txt
```
Should this actually be changed to 3.10.4 as stated in the text of this issue? Or is that not actually a requirement?

**Comment by jprestop** [comment]
Minna had looked into this and found that the highest supported version of Python in ReadtheDocs is Python 3.8. 
Here is a link to the comment where she mentioned in the corresponding [METcalcpy issue](https://github.com/dtcenter/METcalcpy/issues/270):

https://github.com/dtcenter/METcalcpy/issues/270#issuecomment-1416553527

**Comment by georgemccabe** [comment]
The DockerHub build for the 11.1.0-beta1 release failed. Changes in PR #2480 should fix the issue.
